source/m2minterfaceimpl.cpp

```diff
@@ -601,6 +628,7 @@ void M2MInterfaceImpl::state_bootstrap(EventData *data)
         if (error != M2MInterface::ErrorNone) {
             tr_error("M2MInterfaceImpl::state_bootstrap - set error as M2MInterface::InvalidParameters");
             internal_event(STATE_IDLE);
+            strncpy(_error_description,ERROR_REASON_11, sizeof(_error_description));
```

Comment:
Perhaps in future the multiple calls to strncpy() could be replaced with a helper method, which could contain a #ifdef to get rid of the code completely. That would allow linker to remove the error strings too.